sklearn/tree/tests/test_tree.py

```diff
@@ -1266,3 +1268,39 @@ def check_min_weight_leaf_split_level(name):
 def test_min_weight_leaf_split_level():
     for name in ALL_TREES:
         yield check_min_weight_leaf_split_level, name
+
+
+def check_public_apply(name):
+    X_small32 = X_small.astype(tree._tree.DTYPE)
```

Comment:
Is the dtype relevant to the test here? The calls to `check_array` should handle it, right?

Actually, if anything, I think it would be nice to see a test for calling the public API with the wrong dtype (say `int`): if I understand correctly, this should fail if trying to call the underlying `_tree.apply` but work with the public `apply` because of the extra `check_array`.